lib/declare_schema/model/field_spec.rb

```diff
@@ -83,15 +83,18 @@ def initialize(model, name, type, position: 0, **options)
         if @sql_type.in?([:string, :text, :binary, :varbinary, :integer, :enum])
           @options[:limit] ||= Column.native_types[@sql_type][:limit]
         else
-          @options.has_key?(:limit) and raise "unsupported limit: for SQL type #{@sql_type} in field #{model}##{@name}"
+          @sql_type != :decimal && @options.has_key?(:limit) and warn("unsupported limit: for SQL type #{@sql_type} in field #{model}##{@name}")
```

Comment:
One of the benefits of these being exceptions before, is that when running `rails g declare_schema:migration` it would abort the migration entirely with the exception. Is this some way that we could store the state that there are `FieldSpec` errors, and still have the migration generator abort when `FieldSpec.errors.any?`

Reply:
Certainly possible. We could even make it a configurable preference, once we have the preferences convention set up (in an on-deck ticket).

Reply:
I like that idea. If we're okay with just the warnings during migration generation for the time being and want to push the feature request of remembering the errors, and aborting the migration to a later ticket I'm okay with that. This ticket is valuable on its own and doesn't need be blocked by this.

Reply:
Sounds good. I can make a ticket for that future work so we don't forget it.